core/tests/dispatch_integration_tests.rs

```diff
@@ -690,93 +691,87 @@ fn remote_lost_and_continued_connection() {
+    all_pongs_received_future
+        .wait_timeout(PINGPONG_TIMEOUT)
+        .expect("Time out waiting for ping pong to complete");
 
     // Assert that things are going as we expect
-    let pongfn = remote_a.kill_notify(ponger_named);
-    pongfn
-        .wait_timeout(Duration::from_millis(1000))
+    ponger_system_1
+        .kill_notify(ponger_named)
+        .wait_timeout(STOP_COMPONENT_TIMEOUT)
         .expect("Ponger never died!");
+
     pinger_named.on_definition(|c| {
         assert_eq!(c.count, PING_COUNT);
     });
+
     // We now kill system2
-    remote_a.kill_system().ok();
+    ponger_system_1.kill_system().ok();
 
     // Start a new pinger on system
-    let (pinger_named2, pinf2) =
-        system.create_and_register(move || PingerAct::new_lazy(named_path));
-    pinf2.wait_expect(Duration::from_millis(1000), "Pinger failed to register!");
-    system.start(&pinger_named2);
+    let (pinger_named_2, _) = start_pinger(&pinger_system, PingerAct::new_lazy(named_path.clone()));
     // Wait for it to send its pings, system should recognize the remote address
-    thread::sleep(Duration::from_millis(1000));
-    // Assert that things are going as they should be, ping count has not increased
-    pinger_named2.on_definition(|c| {
+    thread::sleep(WAIT_TIME);
+    // Assert that things are going as they should be, pong count has not increased
+    pinger_named_2.on_definition(|c| {
         assert_eq!(c.count, 0);
     });
     // Sleep long-enough that the remote connection will be dropped with its queue
     thread::sleep(Duration::from_millis(
-        3 * 500, // retry config from above
+        CONNECTION_RETRY_INTERVAL * CONNECTION_RETRY_ATTEMPTS as u64,
     ));
     // Start up remote_b
     let mut addr: SocketAddr = "127.0.0.1:0".parse().expect("Address should work");
-    addr.set_port(remote_port);
-    let remote_b = system_from_network_config(NetworkConfig::new(addr));
+    addr.set_port(ponger_system_port);
+    let ponger_system_2 = system_from_network_config(NetworkConfig::new(addr));
 
-    let (ponger_named, ponf) = remote_b.create_and_register(PongerAct::new_lazy);
-    let poaf = remote_b.register_by_alias(&ponger_named, "custom_name");
-    ponf.wait_expect(Duration::from_millis(1000), "Ponger failed to register!");
-    poaf.wait_expect(Duration::from_millis(1000), "Ponger failed to register!");
-    remote_b.start(&ponger_named);
+    let (ponger_named, _) = start_ponger(&ponger_system_2, PongerAct::new_lazy());
+    ponger_system_2
+        .register_by_alias(&ponger_named, "custom_name")
+        .wait_expect(REGISTRATION_TIMEOUT, "Ponger failed to register!");
 
     // We give the connection plenty of time to re-establish and transfer it's old queue
-    thread::sleep(Duration::from_millis(5000));
-    // Final assertion, did our systems re-connect without lost messages?
-    pinger_named2.on_definition(|c| {
+    thread::sleep(WAIT_TIME);
```

Comment:
Instead of waiting, can't we force a message through the queue, so that when it arrives we know that the that queue has been flushed?